src/error/encode.rs

```diff
@@ -1,5 +1,5 @@
 //! Error types associated with encoding to ASN.1 codecs.
-#![expect(missing_docs)]
+#![allow(missing_docs)]
```

Comment:
This should be expect because it should be removed when it is no longer true.

Reply:
It fails the pipeline alone with:

```
error: this lint expectation is unfulfilled
 --> src/error/encode.rs:2:11
  |
2 | #![expect(missing_docs)]
  |           ^^^^^^^^^^^^
  |
  = note: `-D unfulfilled-lint-expectations` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(unfulfilled_lint_expectations)]
  ```
  
  Should we add `#[allow(unfulfilled_lint_expectations)]` as well to the module?

Reply:
I improved the error documentation a bit so that it can be removed. I also removed couple errors since they were not used anymore. There are likely more. Documentation improvements are rather minimal and could be improved in the future.